src/lib/polygon.js

```diff
@@ -174,34 +174,31 @@ polygon.tester = function tester(ptsIn) {
     };
 };
 
+// TODO Somewhat redundant to multiTester in 'lib/select.js'
```

Comment:
> It's not just used by cartesian though, right?

Exactly. E.g. it is used in `polar` and `choropleth`.

> I don't think polygon.multiTester gets called anywhere now

Nope. It's called in `polygon.tester` in the first line.

Reply:
Ideally, all of `cartesian/select.js` should be moved out of `cartesian/` - as @alexcjohnson said it is not _just_ used for cartesian subplots anymore.

What I was suggesting is to move the contents of `lib/select.js` into the same file as `cartesian/select.js` as the contents of `lib/select.js` aren't used anywhere else.

Once that's done, we can move `cartesian/select.js` to either `components/fx` (side-by-side with the hover routines) or in a new `components/` directory e.g. `components/select`.

Reply:
Done. Moved contents of `lib/select.js` into cartesian/select.js`.

Reply:
@rmoestl thanks for moving the contents to `cartesian/select.js`.

Now, would you mind :hocho: the `multiTester` routine and the `TODO` above from this file. `polygon.multiTester` isn't used anywhere now.

Reply:
Sorry, to repeat that, but isnt't `polygon.multiTester` called here https://github.com/plotly/plotly.js/blob/cbd9f0769e69fc0a09ef5415d94eaa167bcf46fd/src/lib/polygon.js#L34?

Reply:
My bad! I grep'ed for `multiTester` not `multitester`. You're correct, `multitester` is currently used in `tester` in  `lib/polygon.js`.

Now, if I recall correctly, that `polygon.multitester` call in `polygon.tester` was only intended for mulit-selections. Git blame points us to https://github.com/plotly/plotly.js/commit/b800e5537fc67d5b0493b33a57418c0a3a2fff9f which is consistent.

When commenting out this line

https://github.com/plotly/plotly.js/blob/cbd9f0769e69fc0a09ef5415d94eaa167bcf46fd/src/lib/polygon.js#L34

no test fail.

So yeah, I think it's safe to :hocho: everything about `multitester` in `lib/polygon.js`.

Reply:
I do think we have enough tests that you could try replacing that line with
`if(Array.isArray(ptsIn[0][0])) throw new Error('multitester called!');`
or something and see if it ever happens.

But if it *does*, could that line just point to the new `multiTester`? Of course doing that would make a circular reference unless these are brought into the same file - which might be better anyway, I feel like these belong together (even if we can't quite agree on where that together should be...)

Reply:
Running all tests didn't throw the error. I left throwing the error in place and added a nice comment to it, so that future readers understand the "why".